src/aui-button/js/aui-button-core.js

```diff
@@ -180,10 +193,29 @@ ButtonExt.prototype = {
      */
     syncButtonExtUI: function() {
         this._uiSetIcon(this.get('icon'));
+        this._uiSetVisibility(this.get('visible'));
         this._setButtonRole();
     },
 
     /**
+     * @method hide
+     * @description Sets the button's `visible` attribute to `false`
+     * @public
+     */
+    hide: function() {
```

Comment:
Same here, widgets already have a `show` and `hide` functions that do exactly what you're doing here. So maybe you can just remove it? Same for button switch.